module-2/backend/week_four.md

```diff
@@ -10,14 +10,49 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+
+A cookie is a way to make HTTP have state. It is something the server gives the client to save locally.
+
 * What’s the difference between a session and a cookie?
+
+A session is a more sophisticated form of cookie. It is encrypted and has somem other friendly Rails features.
+
 * What’s a flash and when do you want to use flashes?
+
+Flashes are popup messages you can set up to show after a certain action, and can be used with conditionals (Failure/success messages)
+
 * Why do people say “HTTP is stateless”?
+
+Every time the client makes a request the history has been cleared and there is no prior history associated with the request.
+
 * What’s authentication? Explain.
+
+Authentication is giving the user a secret token after who they say they are is verified.
+
 * What’s the difference between authentication and authorization?
+
+Authorization is the concept of limiting access for specific roles.
+
 * What’s a before filter?
+
+A before_filter is a helper that executes a method before anything else in that controller/model is instantiated.
```

Comment:
Limited to a controller. Use a similar strategy on models, but those are callbacks, not before filters.